packages/db/src/controller/level.ts

```diff
@@ -175,6 +190,26 @@ export class LevelDbController implements IDatabaseController<Uint8Array, Uint8A
 
     this.metrics?.dbWriteItems.inc({bucket}, itemsRead);
   }
+
+  /** Start interval to capture metric for db size */
+  private collectDbSizeMetric(): void {
+    this.dbSizeMetric();
+    this.dbSizeMetricInterval = setInterval(this.dbSizeMetric.bind(this), DB_SIZE_METRIC_INTERVAL_MS);
+  }
+
+  /** Capture metric for db size */
+  private dbSizeMetric(): void {
+    const minKey = Buffer.from([0x00]);
+    const maxKey = Buffer.from([0xff]);
+
+    this.approximateSize(minKey, maxKey)
+      .then((dbSize) => {
+        this.metrics?.dbSizeTotal.set(dbSize);
+      })
+      .catch(() => {
+        // ignore errors
```

Comment:
@nflaig can you add metrics to track count of errors here + log them to debug? We should never swallow errors. Also, can you track in a histogram the time it takes to run approximateSize?

Reply:
The only reason this would throw an error is if the database is closed which only happens if the process is shutdown. We also clear the interval to call `approximateSize` before that so its really unlikely to happen, also wouldn't care at that point.

> We should never swallow errors

Yes, I agree with this in general but the db package does not use the logger at the moment and I did not want to introduce it just for this case.

> histogram the time it takes to run approximateSize

Was considering that but after looking at the time this takes even for "big" DB sizes I went with "we don't need it"

Let's add a histogram to track the time and also set a `status` label with `success` and `error` value then we see if this ever throws an error during normal operation.

Reply:
Re logger: then you can move the metric caller outside, so that the interval is controlled from somewhere else which has access to the logger. 

Re metrics: would prefer to have the histogram tracking time for all cases, and the error counter as a separate.

Reply:
> Re logger: then you can move the metric caller outside, so that the interval is controlled from somewhere else which has access to the logger.

If we really want a debug error log here which (I still think we don't need it), then I would rather go with injecting the logger to `LevelDbController`, this is the pattern we usually use. In my opinion it is not a good idea to move the logic to track the db size outside of this class/file.

> Re metrics: would prefer to have the histogram tracking time for all cases, and the error counter as a separate.

sure sounds good as well

Reply:
implemented in https://github.com/ChainSafe/lodestar/pull/5190